### ISSUE TITLE
System.OverflowException in GetPointerId

### ISSUE THREAD

**Issue by vpenades** [MonoGame/MonoGame#6839]
We just got a report from a client that got this exception:

```
System.OverflowException: Arithmetic operation resulted in an overflow.
   en Microsoft.Xna.Framework.Windows.MessageExtensions.GetPointerId(Message msg)
   en Microsoft.Xna.Framework.Windows.WinFormsGameForm.WndProc(Message& m)
   en System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

Of interest is the fact that our client is using a wide table touch projector, so my guess is that the table touch driver is converting the touch events into mouse click events or something like that.

#### What version of MonoGame does the bug occur on:
- MonoGame 3.7.0.1708

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- WindowsDX

**Comment by intelligentvending** [comment]
I have encountered this issue under the following circumstances:

* Using MonoGame 3.8.0.1641, WindowsDX
* Using a touch panel
* Targeting x64 platform
* Using a secondary display with a negative height in respect to the main display

The problem does not occur if targeting x86 or if the secondary display's height is equal to or lower, relative to the main display:

E.g. (note: screen #2 in these images is the primary display)

This is fine:
![image](https://user-images.githubusercontent.com/5197431/217511519-d243e84a-71d1-4fcd-ac9f-b4cc292f0788.png)

This is not:
![image](https://user-images.githubusercontent.com/5197431/217511466-c109d651-a1e2-4007-aa46-9d49ce99922f.png)

Coincidentally, I am using my own user32 interop code to capture touch inputs and received the same exception under these circumstances, with the error occurring while casting the low/high values from the Message.LParam:

        int Low16(IntPtr value)
        {
            return unchecked((short)(uint)value);   // classic unchecked cast to uint
        }

Changing the uint cast to Uint64 stopped the exception on this line, however it was then immediately thrown by MonoGame (which caused some confusion!), so I am unable to confirm if this would actually fix the issue, but it might offer some helpful clues etc.

TLDR; it appears this may be a marshaling issue under x64 when dealing with touch points with a negative value.
